src/Microsoft.SymbolStore/SymbolStores/HttpSymbolStore.cs

```diff
@@ -75,9 +79,26 @@ public void ResetClientFailure()
         protected override async Task<SymbolStoreFile> GetFileInner(SymbolStoreKey key, CancellationToken token)
         {
             Uri uri = GetRequestUri(key.Index);
-            Stream stream = await GetFileStream(uri, token);
+
+            Stream stream;
+            bool needsChecksumMatch = key.PdbChecksums.Any();
+
+            if (needsChecksumMatch)
+            {
+                string checksumHeader = string.Join(";", key.PdbChecksums);
+                HttpClient client = _authenticatedClient ?? _client;
+                Tracer.Information($"SymbolChecksum: {checksumHeader}");
+                _client.DefaultRequestHeaders.Add("SymbolChecksum", checksumHeader);
+            }
+
+            stream = await GetFileStream(uri, token);
             if (stream != null)
             {
+                if(needsChecksumMatch)
+                {
+                    var validator = new ChecksumValidator(_tracer);
```

Comment:
maybe I should just make this static

Reply:
Sounds good.  Minor nit: there needs to be a space after the if (needsChecksumMatch)